apps/pipelines/nodes/base.py

```diff
@@ -23,71 +20,62 @@ def add_messages(left: list, right: list):
 
 class PipelineState(TypedDict):
     messages: Annotated[Sequence[Any], add_messages]
+    experiment_session: ExperimentSession
 
 
 class PipelineNode(BaseModel, ABC):
-    """Pipeline node that is either a single or a combination of Langchain Runnables
-
-    Define required parameters as typed fields. Compose the runnable in `get_runnable`
+    """Pipeline node that implements the `_process` method and returns a new state. Define required parameters as
+    typed fields.
 
     Example:
         class FunNode(PipelineNode):
             required_parameter_1: CustomType
             optional_parameter_1: Optional[CustomType] = None
 
-            def get_runnable(self, input) -> str:
-                if self.optional_parameter_1:
-                    return PromptTemplate.from_template(template=self.required_parameter_1)
-                else:
-                    return LLMResponse.build(node)
+            def _process(self, state: PipelineState) -> PipelineState:
+                input = state["messages"][-1]
+                output = ... # do something
+                return output # The state will be updated with output
 
        class FunLambdaNode(PipelineNode):
             required_parameter_1: str
 
-            def get_runnable(self, node: Node):
-                def fn(input: Input):
-                    return self.required_parameter_1
-                return RunnableLambda(fn, name=self.__class__.__name__)
+            def _process(self, state: PipelineState) -> PipelineState:
+                ...
+                return # The state will not be updated, since None is returned
+
     """
 
+    _config: RunnableConfig | None = None
+
     class Config:
         arbitrary_types_allowed = True
 
     @classmethod
     def build(cls, node: Node) -> Callable[[dict], dict]:
```

Comment:
I don't think this is used anywhere now?

Reply:
correct, I'll remove